### ISSUE TITLE
[BUG] invalid keras import in tests (for some versions)

### ISSUE THREAD

**Issue by ltsaprounis** [sktime/sktime#3743]
It seems like there's some inconsistencies with keras imports between keras versions.
Looking at the toml file I don't see any bounds on the versions. 

In any case I'm on keras 2.8 and the following import in sktime/classification/tests/test_base.py raises an import error:
`from keras.optimizers import Adamax`

This causes the tests to fail during collection, which is annoying (especially for the VS code testing functionality)
to test that without running all the tests you can do: `pytest --collect-only .`

doing some debugging I found that the right import for my case is: 
`from keras.optimizer_v2.adamax import Adamax`

The 2 ways to resolve this (I suggest we do both):
- Add version bounds to avoid this in the future (also why are we using keras and not tensorflow.keras ?)
- move the import inside the test as it's a soft dependency anyway

**Comment by fkiraly** [comment]
hm, but moving the import inside the test wouldn't solve the issue when executing the test?

Third option: check the version bound where the import changes, and have if/else conditional import

**Comment by ltsaprounis** [comment]
> this should also fail in #2986
> 
> we should run the diagnostic more often.

Is it because soft-dependency imports should be inside the test?

**Comment by fkiraly** [comment]
no, even if the soft dep imports are inside the tests, the diagnostic #2986 would still fail, but it would appear as a test failure (because you are then also testing the import).

The way to isolate the soft deps is have a `pytest.skipif` depending on package availability or estimator support, search for `.skipif` to find examples in the code base.

**Comment by fkiraly** [comment]
couple unisolated imports have crept in since we last did this...

We should fix, opened an issue:
https://github.com/sktime/sktime/issues/3748
